sdk/opa.go

```diff
@@ -508,6 +535,9 @@ func evaluate(ctx context.Context, args evalArgs) (interface{}, ast.Value, map[s
 		rego.EvalInterQueryBuiltinCache(args.interQueryCache),
 		rego.EvalNDBuiltinCache(args.ndbcache),
 		rego.EvalQueryTracer(args.tracer),
+		rego.EvalMetrics(args.m),
```

Comment:
Question: would it ever make sense to allow a user to set a different metrics instance for evaluation? `opt eval` uses the same one so I've gone with that for now.

Reply:
In general? If you hold on to the prepared eval query, it would make sense to have new metrics objects for the (potentially) multiple _actual evals_, I think.

 If it's all a one-off eval, like `opa eval`, it wouldn't be necessary, since the recorded metrics are going to be disjunct anyways (I think).

Reply:
More specifically in this case. Should [`evaluate`](https://github.com/open-policy-agent/opa/blob/main/sdk/opa.go#L491) ever need to support different metrics sets for `PrepareForEval` and `Eval`? It seems unlikely...

Reply:
I don't see any code paths here where you can hit one but not the other, so it ought to be safe to roll the two together. In the future, if we want to split the measurements apart, we can readily do that, although it might require careful renaming of metrics. :thinking: